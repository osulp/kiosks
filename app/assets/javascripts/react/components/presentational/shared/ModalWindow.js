import {connect} from 'react-redux';
import React, {Component, PropTypes} from 'react';

class ModalWindow extends Component {
  closeModal(event) {
    // prevents clicks from within the root_component from triggering the close. User can click on the close
    // icon or anywhere outside of the root_component to close the modal.
    if(event.target.id == "modal" || event.target.id == "modal_close") {
      this.props.setModalRootComponent();
      this.props.setModalVisibility(false);
    }
  }

  render() {
    let display = this.props.visible ? "block" : "none";
    let root_component = this.props.root_component;
    let close_icon_component = React.createElement(
      "SPAN",
      {
        className: "glyphicon glyphicon-remove-circle",
        onClick: this.closeModal.bind(this),
        id: "modal_close"
      },
      "");
    return (
      <div id="modal" style={{display: display}} onClick={this.closeModal.bind(this)}>
        {root_component ? close_icon_component : undefined }
        {root_component}
      </div>
    );
  }
}

ModalWindow.propTypes = {
  visible: PropTypes.bool.isRequired,
  root_component: PropTypes.element,
  setModalVisibility: PropTypes.func.isRequired,
  setModalRootComponent: PropTypes.func.isRequired,
};

export default ModalWindow;
