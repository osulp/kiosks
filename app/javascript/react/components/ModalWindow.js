import { bindActionCreators } from "redux"
import { connect } from "react-redux"

import ModalWindow from "./presentational/shared/ModalWindow"
import * as actions from "../actions"

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = state => {
  return {
    visible: state.modal.visible,
    root_component: state.modal.root_component
  }
}

// Map the actions to the dispatcher
export const mapDispatchToProps = dispatch => {
  return bindActionCreators(actions, dispatch)
}

// Connect the mappings (state -> properties, and actions -> dispatch) to the component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModalWindow)
