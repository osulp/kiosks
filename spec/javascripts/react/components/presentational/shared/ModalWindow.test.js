import React from 'react';
import ModalWindow from '../../../../../../app/assets/javascripts/react/components/presentational/shared/ModalWindow';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const setup = (args) => {

  const props = {
    visible: args.visible,
    root_component: args.element,
    setModalVisibility: jest.fn(),
    setModalRootComponent: jest.fn()
  };

  const component = renderer.create(
    <ModalWindow {...props} />
  );

  return { props, component };
};

describe('Snapshot', () => {
  it('matches the cached snapshot', () => {
    let {component} = setup({visible: true, element: React.createElement("H1", {}, "test")});
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Shared::ModalWindow', () => {
  it('logs a console.error when visible with no root_component', () => {
    global.console = jest.fn();
    global.console.error = jest.fn();
    let { component } = setup({visible: true, element: undefined });
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(global.console.error.mock.calls.length).toEqual(1);
  });
  it('can be closed', () => {
    let {component, props} = setup({visible: true, element: React.createElement("H1", {}, "test")});
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.onClick({target: {id: "modal"}});
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(props.setModalRootComponent.mock.calls.length).toEqual(1);
    expect(props.setModalVisibility.mock.calls.length).toEqual(1);
  });
});

