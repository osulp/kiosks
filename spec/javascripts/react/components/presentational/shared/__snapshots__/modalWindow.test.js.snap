exports[`Shared::ModalWindow can be closed 1`] = `
<div
  id="modal"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
    }
  }>
  <SPAN
    className="glyphicon glyphicon-remove-circle"
    id="modal_close"
    onClick={[Function]}>
    
  </SPAN>
  <H1>
    test
  </H1>
</div>
`;

exports[`Shared::ModalWindow can be closed 2`] = `
<div
  id="modal"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
    }
  }>
  <SPAN
    className="glyphicon glyphicon-remove-circle"
    id="modal_close"
    onClick={[Function]}>
    
  </SPAN>
  <H1>
    test
  </H1>
</div>
`;

exports[`Snapshot matches the cached snapshot 1`] = `
<div
  id="modal"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
    }
  }>
  <SPAN
    className="glyphicon glyphicon-remove-circle"
    id="modal_close"
    onClick={[Function]}>
    
  </SPAN>
  <H1>
    test
  </H1>
</div>
`;
